src/subtype.c

```diff
@@ -2389,8 +2389,16 @@ static jl_value_t *bound_var_below(jl_tvar_t *tv, jl_varbinding_t *bb, jl_stenv_
             return bb->lb;
         return jl_box_long(blb - offset);
     }
-    if (offset > 0)
-        return NULL;
+    if (offset > 0) {
```

Comment:
Is it worthwhile checking `e->invdepth` here also, to try to avoid making the new TypeVar when it is in covariant context anyways?

Reply:
`bound_var_below` is only used in `intersect_varargs` thus it's always in invariant context?
And the `invdepth` has been checked above in L2377
```julia
    if (bb->depth0 != e->invdepth)
        return jl_bottom_type;
```

Reply:
Ah, okay, that seems good then.